sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -299,7 +303,16 @@ fun ObservableInterface.unsubscribeSourceRemoved(observer: Observer) =
  */
 fun Event.getResourceEventData(): ResourceEventData {
   val json = data.toJson()
-  return Gson().fromJson(json, ResourceEventData::class.java)
+  return gson.fromJson(json, ResourceEventData::class.java)
```

Comment:
should we add try-catch block?

Reply:
imho if the data format has been changed, it should be a bug and should throw exceptions so that we can fix it in a patch.

Reply:
we can discuss it with the team. probably leave as is for now. but let's think about it

several options for now:

- return nullable
- support some default value and work with it somehow

also we can send telemetry event for such crashes

Reply:
Sounds good, let's make it as a follow up 👍

Reply:
> yeah, but is it proper behavior for users? is it possible that our users (developers) can release some app in play market and got crash in app?

That statement could be true for every line of code we ship. The program is expected to return this non null data in that specific format, it's part of the contract that gl-native makes with us. Overall, I'm personally leaning more towards a crash-fast approach, we catch issues during development, else they can become unnoticed and broken if not properly exercised. Additionally introducing null, will require developers to handle null which add logical branching to their code. This makes it more complex to handle, while it's not expected program output.